### ISSUE TITLE
`next(iter(loader))` inside eval loops rebuilds the iterator every iteration

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#27]
```python
# scripts/modeling/hits_encdec/eval.py:368-373
for _ in range(n_eval_iters):
    src, ctx, tgt = parse_batch(next(iter(train_loader)), device)
```

`next(iter(loader))` builds a **new** iterator every iteration, which re-shuffles the sampler and (with `num_workers > 0`) re-spawns workers, just to read the first batch. With `n_eval_iters: 100` that's 100 iterator constructions per eval.

It also means eval draws 100 random batches with replacement rather than iterating the split, so "val loss" is a noisy estimate over an arbitrary subset instead of the actual validation loss.

Same pattern in: `encdec/eval.py`, `chroma/eval.py`, `hits_encdec_polydrums/train.py:438,448`, `hits_dec/train.py:139,147`, `hits_dec_xformer/train.py:129`, `rsp_fc/train.py:91,103`, `tvae/train.py:125`, `dec/train.py:108,116`, and the `eval_gen_*` while-loops that collect inference sources.

**Fix**: create the iterator once outside the loop (`it = iter(loader)`), or just iterate the loader and break after `n_eval_iters` batches. Ideally evaluate over the whole val split and report a real mean.
